### PR TITLE
feat: demolish building

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/BuildingRegistry/Building/BuildingUnit.cs [hunk 1]
```diff
@@ -432,6 +432,40 @@ public void DetachAddressBecauseAddressWasRemoved(AddressPersistentLocalId addre
...
         }
 
+        public void Demolish()
+        {
+            if (IsRemoved)
+            {
+                return;
+            }
+
+            if (Status == BuildingUnitStatus.NotRealized || Status == BuildingUnitStatus.Retired)
```
Comment (emalfroy) [inline src/BuildingRegistry/Building/BuildingUnit.cs:442]: Do we have a guard status?

File: src/BuildingRegistry.Api.BackOffice/Building/BuildingController-Demolish.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        /// <param name="ifMatchHeaderValue"></param>
+        /// <param name="cancellationToken"></param>
+        [HttpPost("{persistentLocalId}/acties/slopen")]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+        [SwaggerResponseHeader(StatusCodes.Status202Accepted, "location", "string", "De url van het gebouw.")]
+        [SwaggerResponseExample(StatusCodes.Status400BadRequest, typeof(BadRequestResponseExamples))]
+        [SwaggerResponseExample(StatusCodes.Status500InternalServerError, typeof(InternalServerErrorResponseExamples))]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = PolicyNames.GeschetstGebouw.DecentraleBijwerker)]
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice/Building/BuildingController-Demolish.cs:39]: No Authorization.

**Review by emalfroy — APPROVED** [review]

File: src/BuildingRegistry.Api.BackOffice/Building/BuildingController-Demolish.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        public async Task<IActionResult> Demolish(
+            [FromServices] BuildingExistsValidator buildingExistsValidator,
+            [FromServices] IIfMatchHeaderValidator ifMatchHeaderValidator,
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice/Building/BuildingController-Demolish.cs:42]: We don't do if match header validation as request are done by ourselves based upon lines in the uploaded grb file.

File: src/BuildingRegistry/Building/Building.cs [hunk 4]
```diff
@@ -236,6 +236,30 @@ public void ChangeOutliningConstruction(ExtendedWkbGeometry extendedWkbGeometry)
...
+            if (BuildingStatus == BuildingStatus.Retired)
+            {
+                return;
+            }
+
+            GuardValidStatusses(BuildingStatus.Realized);
+
+            ApplyChange(new BuildingWasDemolished(BuildingPersistentLocalId));
+
+            foreach (var unit in _buildingUnits)
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building.cs:257]: I think the building unit events should come before the building event, see
https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry/Building/Building.cs#L99-L104
and
https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry/Building/BuildingUnit.cs#L87-L95

File: src/BuildingRegistry/Building/BuildingUnit.cs (hunk 1 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/BuildingUnit.cs:442]: A guard will throw an exception. Here we don't want to throw an exception, but rather do nothing. So this seems ok.

**Review by jvandaal — DISMISSED** [review]

**Review by emalfroy — DISMISSED** [review]

File: src/BuildingRegistry/Building/Building.cs [hunk 5]
```diff
@@ -236,6 +236,30 @@ public void ChangeOutliningConstruction(ExtendedWkbGeometry extendedWkbGeometry)
                 buildingUnitsPosition));
         }
 
+        public void Demolish()
```
Comment (ArneD) [inline src/BuildingRegistry/Building/Building.cs:239]: Can you move this to `Building_Grb` ?

File: src/BuildingRegistry/Building/BuildingUnit.cs [hunk 6]
```diff
@@ -432,6 +432,40 @@ public void DetachAddressBecauseAddressWasRemoved(AddressPersistentLocalId addre
                 newAddressPersistentLocalId));
         }
 
+        public void Demolish()
```
Comment (ArneD) [inline src/BuildingRegistry/Building/BuildingUnit.cs:435]: Maybe we should also create a partial class here, it's a long file :) ? 
`BuildingUnit_Grb` ?

File: src/BuildingRegistry/Building/Building.cs [hunk 7]
```diff
@@ -236,6 +236,30 @@ public void ChangeOutliningConstruction(ExtendedWkbGeometry extendedWkbGeometry)
...
+            }
+
+            GuardValidStatusses(BuildingStatus.Realized);
+
+            foreach (var unit in _buildingUnits)
+            {
+                unit.Demolish();
+            }
+
+            ApplyChange(new BuildingWasDemolished(BuildingPersistentLocalId));
```
Comment (ArneD) [inline src/BuildingRegistry/Building/Building.cs:260]: Don't forget the GRB event.

**Review by ArneD — APPROVED** [review]

**Pull request merged**